### PR TITLE
Add test suite for Plugins folder

### LINKED ISSUE #934: Introduce `plugins` folder for standalone plugins
This issue is part 1/2 for setting up the new infrastructure for developing standalone plugins.

## Requirements

**This work should be implemented against a new `feature/modules-to-plugins` feature branch.**

* Create new `plugins` folder in the repository root.
* Add a `.gitkeep` to it.
* Add the `plugins` folder to be ignored in `.gitattributes` so that it won't be part of the Performance Lab plugin build.
* Restructure the `plugins.json` file:
    * Going forward, there should be always two top level properties: `modules` and `plugins`.
    * `modules` should have the map of "module => plugin data (slug and version)", i.e. the same thing that the file contains today.
    * `plugins` should have an array of plugin slugs.
* Update the existing workflows that use the `plugins.json` file to look for modules within the new `modules` property in the file.
* Add references to the new `plugins` folder in any dev tooling that currently lints/covers the `modules` folder (e.g. in `phpcs.xml.dist` etc.).

Note: Additional changes to build/publish the new plugins should be implemented in a separate follow-up issue.

### PR DESCRIPTION
**Pull request by mukeshpanchal27**
## Summary

Fixes #934

**NOTE: Checking test suite for plugins folder.**

## Checklist

- [x] PR has either `[Focus]` or `Infrastructure` label.
- [x] PR has a `[Type]` label.
- [x] PR has a milestone or the `no milestone` label.

### CODE REVIEW

**Comment by swissspidy** [comment]
```
> wp-env run tests-cli --env-cwd=/var/www/html/wp-content/plugins/$(basename $(pwd)) composer test -- --testsuite=webp-uploads

ℹ Starting 'composer test --testsuite=webp-uploads' on the tests-cli container. 
```

Soo... four dashes it is? 🤔  Unless wp-env has some weird logic for passing additional args.

```
$ npm run test-php -- -- -- --testsuite=webp-uploads

> test-php
> wp-env run tests-cli --env-cwd=/var/www/html/wp-content/plugins/$(basename $(pwd)) composer test -- -- --testsuite=webp-uploads
```

This should run `composer test -- --testsuite=webp-uploads` on the container, which would then run phpunit correctly.

**Comment by thelovekesh** [comment]
@mukeshpanchal27 @swissspidy I was wondering if we can avoid the `wp-env` based WP setup in CI and instead use SVN+Maridb to set up the test suite?

With `wp-env`:
- we can see increased CI times(on AMP we reduced it to 3-4 mins from 7-8 mins).
- API changes. The previous one was related to removing the PHPUnit containers.
- Maintenance issues in the case of the above scenario.

On the AMP plugin, we faced similar issues and we had to remove `wp-env` and switch back to the "vanilla" setup for multisite tests.

Recently I was working on a POC to improve the workflows for the Performance Lab plugin which can be found at - https://github.com/thelovekesh/performance/pull/2 and it uses bash-based tests setup.

Also, let me know if the above PR makes sense, and I will open it on this repo after testing.

**Comment by swissspidy** [comment]
I see your point and personally like not having to use wp-env on CI. There are pros and cons for each setup though. Probably best to open an issue for wider discussion.

**Comment by mukeshpanchal27** [comment]
@swissspidy, do we need to merge `trunk` into `feature/modules-to-plugins` to get the unit tests related changes in the branch, and then rebase this PR branch onto `feature/modules-to-plugins`? What do you think?

**Comment by swissspidy** [comment]
Seems like it 👍

**Comment by mukeshpanchal27** [comment]
> @swissspidy, do we need to merge `trunk` into `feature/modules-to-plugins` to get the unit tests related changes in the branch, and then rebase this PR branch onto `feature/modules-to-plugins`? What do you think?

PR #959.

**Comment by mukeshpanchal27** [comment]
The PR currently indicates failures in certain workflows, even though these issues have already been addressed in the trunk. The changes from #968 will be incorporated, and once completed, the PR should function correctly.

**Review by swissspidy — APPROVED** [review]

File: tests/plugins/webp-uploads/test-plugins.php
Comment (westonruter) [inline tests/plugins/webp-uploads/test-plugins.php:1]: Question: Why not move the existing tests from `tests/modules/images/webp-uploads`?

**Review by westonruter — APPROVED** [review]

File: tests/plugins/webp-uploads/test-plugins.php
Comment (mukeshpanchal27) [reply tests/plugins/webp-uploads/test-plugins.php:1]: In the https://github.com/WordPress/performance/pull/956/commits/500fb914f2e7998e8435cb22bf0dcc2ee6bc7477, I have updated the plugin name to "Test Plugin." Once we have an actual plugin in the plugins folder, and the infrastructure is set up, we will remove this test plugin.

**Pull request merged**